src/main/java/io/github/thebusybiscuit/cscorelib2/inventory/InvUtils.java

```diff
@@ -56,7 +77,7 @@ public static boolean fits(@NonNull Inventory inv, @NonNull ItemStack item, int.
 
             if (stack == null || stack.getType() == Material.AIR) {
                 return true;
-            } else if (stack.getAmount() + item.getAmount() <= stack.getMaxStackSize() && ItemUtils.canStack(stack, item)) {
+            } else if (underMaxStackSize(stack, item, inv) && ItemUtils.canStack(stack, item)) {
```

Comment:
Need to update that too
```suggestion
            } else if (isValidStackSize(stack, item, inv) && ItemUtils.canStack(stack, item)) {
```